scripts/missions/asa/helpers.lua

```diff
@@ -0,0 +1,39 @@
+-----------------------------------
+-- A Shantotto Ascension Helpers
+-----------------------------------
+require('scripts/globals/missions')
+-----------------------------------
+
+xi = xi or {}
+xi.asa = xi.asa or {}
+xi.asa.helpers = xi.asa.helpers or {}
+
+local hp = {80, 60, 40, 20, 1}
+
+local abilities =
+{
+    { 7041, 893 }, -- Ramuh
+    { 7042, 857 }, -- Titan
+    { 7043, 848 }, -- Ifrit
+    { 7044, 866 }, -- Leviathan
+    { 7045, 875 }, -- Garuda
+    { 7046, 884 }, -- Shiva
+}
+
+xi.asa.helpers.astralFlow = function(mob)
+    local astralFlows = mob:getLocalVar("astralflows")
+    for k, v in pairs(hp) do
+        if astralFlows == k and mob:getHPP() <= v and mob:canUseAbilities() then
+            mob:setLocalVar("astralflows", astralFlows + 1)
+            for _, v in pairs(abilities) do
```

Comment:
instead of a loop looking for the correct lookup in the table, just lookup straight into a table:
```lua
local abilities =
{
    [7041] = 893, -- Rumah
...


mob:useMobAbility(abilities[mob:getPool()])

Removing that loop will also stop CI complaining about shadowing inner/outer vars

Reply:
Thanks for the advice. Got that adjusted to use your recommendation instead.